cache_system/src/backend/shared.rs

```diff
@@ -24,9 +26,21 @@ where
     V: Clone + Debug + Send + 'static,
 {
     /// Create new backend around the inner backend
-    pub fn new(inner_backend: Box<dyn CacheBackend<K = K, V = V>>) -> Self {
+    pub fn new(
+        inner_backend: Box<dyn CacheBackend<K = K, V = V>>,
+        name: &'static str,
+        metric_registry: &metric::Registry,
+    ) -> Self {
+        let metric_removed_by_predicate = metric_registry
+            .register_metric::<U64Counter>(
+                "cache_removed_by_predicate",
+                "Number of entries removed from a cache via a filter predicate",
```

Comment:
For someone unfamiliar with how this code works, it may not be clear what a `'filter predicate'` is. 

```suggestion
                "Number of entries removed from a cache via a custom condition",
```

I am not sure 'custom' is really all that much better than `filter` 🤔

Reply:
:+1: